generators/client/templates/react/src/main/webapp/app/config/translation.ts

```diff
@@ -1,10 +1,11 @@
 /* global: require*/
 import { TranslatorContext, Storage } from 'react-jhipster';
+import { merge } from 'lodash';
 
 import { setLocale } from '../reducers/locale';
 
 const mergeTranslations = requireContext => requireContext.keys().reduce(
-  (merged, key) => ({ ...merged, ...requireContext(key) }),
+  (merged, key) => merge({ ...merged }, { ...requireContext(key) }),
```

Comment:
@qmonmert this will solve the i18n issue you encountered

Reply:
Cant we do this with `Object.assign` do we need `lodash` for this 🤔

Reply:
Object.assign is not recursive, its same as { ... } destructuring which is only first level, we need deep merge here